### DISCUSSION TITLE
iPXE Shell UEFI Boot

### DISCUSSION THREAD

**Discussion by SpicyLew12** [ipxe/ipxe#197] [Q&A] [answered]
I am trying to boot into the iPXE shell from a CDROM, this is easy in legacy mode as all I need to do is burn the ipxe.iso to a disk `cdrecord -v -dao ipxe.iso`
is it possible to boot into iPXE shell from a CDROM when booting in UEFI mode (isohybrid). I cannot use a USB drive and am on an airgapped network(no access to internet, no DHCP).

**Comment by mcb30** [comment, marked as answer]
There's an open issue #131 to provide the feature that you're looking for - you may want to track that issue

**Reply by SpicyLew12** [reply]
yes that looks like the feature I need, I will try to homebrew my own iso in the meantime

**Reply by SpicyLew12** [reply]
So I was actually able to successfully create a homebrewed isohybrid the build is as follows:
deconstruct the ipxe.iso insert the ipxe.efi generate a ESP image and regenerated the iso with both options --steps may need to change with architecture 
`mkdir -p iso/efi/boot`
`sudo mount ipxe.iso /mnt`
`cp  /mnt/*  iso/`
`cp ipxe.efi  iso/efi/boot/bootx64.efi`
`sz=$(du -s iso/)`
`truncate -s) $(($sz+128+128))k esp.img`
`mkfs.msdos -F 12 -f 1 -r 112 -R 1 esp.img`
`sudo umount /mnt`
`sudo mount esp.img /mnt -o uid=whoami `
`cp -av iso/* /mnt/`
`sudo umount /mnt`
`cp esp.img iso/efi`
`mkisofs -o iPXEhybrid.iso -b  isolinux.bin -c boot.cat -no-emul-boot -boot-load-size 4 -boot-info-table -eltorito-alt-boot -e efi/esp.img -no-emul-boot -J -R -v -T -V 'IPXE' iso/`

**Reply by mcb30** [reply]
@SpiceyLew12 issue #131 is now fixed: you should be able to use just `make bin-x86_64-efi/ipxe.iso`

You can also now create a hybrid BIOS+UEFI ISO image by invoking the `util/genfsimg` script manually.  For example:

```
make bin/ipxe.lkrn bin-x86_64-efi/ipxe.efi
./util/genfsimg -o hybrid.iso bin/ipxe.lkrn bin-x86_64-efi/ipxe.efi
```
